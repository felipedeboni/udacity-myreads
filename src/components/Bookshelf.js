import React from 'react';
import PropTypes from 'prop-types';
import Book from './Book';

const Bookshelf = ({ id, title, onMoveToShelf, books }) => {
  const booksOnShelf = books.filter(book => book.shelf === id);

  return (
    <div className="bookshelf">
      <h2 className="bookshelf-title">{title}</h2>
      <div className="bookshelf-books">
        <ol className="books-grid">
          {booksOnShelf.map(book => (
            <li key={book.id}>
              <Book book={book} onMoveToShelf={onMoveToShelf} />
            </li>
          ))}
        </ol>
      </div>
    </div>
  );
};

Bookshelf.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  books: PropTypes.array.isRequired,
  onMoveToShelf: PropTypes.func.isRequired
};

export default Bookshelf;
