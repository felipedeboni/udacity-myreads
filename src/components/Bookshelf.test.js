import React from 'react';
import Bookshelf from './Bookshelf';
import { shallow } from 'enzyme';

const books = [
  {
    id: 'IOejDAAAQBAJ',
    shelf: 'currentlyReading'
  },
  {
    id: 'sJf1vQAACAAJ',
    shelf: 'currentlyReading'
  },
  {
    id: 'xlp6NE2NWecC',
    shelf: 'wantToRead'
  }
];

it('renders without crashing', () => {
  const wrapper = shallow(
    <Bookshelf
      id="currentlyReading"
      title="Currently Reading"
      books={books}
      onMoveToShelf={jest.fn()}
    />
  );

  expect(wrapper).toMatchSnapshot();
});

it('should only show his books', () => {
  const wrapper = shallow(
    <Bookshelf
      id="currentlyReading"
      title="Currently Reading"
      books={books}
      onMoveToShelf={jest.fn()}
    />
  );
  expect(wrapper.find('Book').length).toBe(2);
});
