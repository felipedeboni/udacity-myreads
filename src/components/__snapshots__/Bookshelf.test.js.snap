// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="bookshelf"
>
  <h2
    className="bookshelf-title"
  >
    Currently Reading
  </h2>
  <div
    className="bookshelf-books"
  >
    <ol
      className="books-grid"
    >
      <li
        key="IOejDAAAQBAJ"
      >
        <Book
          book={
            Object {
              "id": "IOejDAAAQBAJ",
              "shelf": "currentlyReading",
            }
          }
          height={192}
          hideNone={false}
          onMoveToShelf={[Function]}
          width={128}
        />
      </li>
      <li
        key="sJf1vQAACAAJ"
      >
        <Book
          book={
            Object {
              "id": "sJf1vQAACAAJ",
              "shelf": "currentlyReading",
            }
          }
          height={192}
          hideNone={false}
          onMoveToShelf={[Function]}
          width={128}
        />
      </li>
    </ol>
  </div>
</div>
`;
